modules/kubernetes-addons/cluster-autoscaler/locals.tf

```diff
@@ -56,15 +35,16 @@ locals {
     }
   ]
 
-  default_helm_values = [templatefile("${path.module}/values.yaml", {
-    aws_region           = var.addon_context.aws_region_name,
-    eks_cluster_id       = var.addon_context.eks_cluster_id
-    service_account_name = local.service_account_name
-  })]
+  irsa_config = {
+    create_kubernetes_namespace       = true
+    kubernetes_namespace              = local.name
+    create_kubernetes_service_account = true
+    kubernetes_service_account        = local.service_account_name
+    irsa_iam_policies                 = [aws_iam_policy.cluster_autoscaler.arn]
```

Comment:
Add `var.irsa_permission_boundary` as input and pass it in `irsa_config` in case customers want to pass it in.

Reply:
I will add this to my Addon upgrade PR. Merging this now